spec/controllers/users_controller_spec.rb

```diff
@@ -70,4 +108,29 @@
       expect(assigns(:user)).to eq(user)
     end
   end
-end
\ No newline at end of file
+
+  describe "GET new_with_invitation_token" do
+    it "renders the :new view template" do
+      invitation = Fabricate(:invitation)
+      get :new_with_invitation_token, token: invitation.token
+      expect(response).to render_template :new
+    end
+
+    it "sets @invitation_token" do
+      invitation = Fabricate(:invitation)
+      get :new_with_invitation_token, token: invitation.token
+      expect(assigns(:invitation_token)).to eq(invitation.token)
+    end
+
+    it "renders the new user page" do
+      invitation = Fabricate(:invitation)
+      get :new_with_invitation_token, token: invitation.token
+      expect(response).to render_template :new
+    end
+
+    it "redirects to expired token page for invalid token" do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.